### PR TITLE
FITTFinder community story

### PR DESCRIPTION
**Pull request by mooreds**

### CODE REVIEW

**Review by jobannon — APPROVED** [review]
Interesting article.  Mostly just comments.  I wish that we had a response callout that could offer tips about things we are doing to mitigate or at least consider the issues raised.

BTW, looks like their site is down.  Tough day I suppose.

File: site/_posts/2021-01-26-fitt-finder-fusionauth.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+**Dan:** Did you build your own user interface to the login flows, or are you using themed pages?
+
+**Trevor:** We are currently using our own very simple login page. I need to set up themed pages for the rest of the flows before we launch. We may keep our own basic login UI though, so users can login within the app instead of redirecting.
+
+> FusionAuth replaced some custom user management and authentication done by a prior contractor before my involvement. 
+
+**Dan:** What are your plans for social sign-on and SAML SSO?
+
+**Trevor:** We don't want our users to have to create passwords if they're comfortable using social sign-on or they're coming from a third-party using SSO. For social, we've already enabled Google, and we'll likely add Apple and Facebook. We're in the process of configuring SAML v2 for the first time, but ran into a snag because FusionAuth doesn't accept the partner's 1024-bit RSA keys. Hopefully they're able to use a stronger key.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/_posts/2021-01-26-fitt-finder-fusionauth.md:42]: great catch, Daniel fixed this in https://github.com/FusionAuth/fusionauth-issues/issues/1091

File: site/_posts/2021-01-26-fitt-finder-fusionauth.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+> We eventually want to scale to millions of consumers, many of whom will likely use our service for free, so the per-user costs of other auth platforms seemed prohibitive, especially for a bootstrapped start-up.
+
+**Dan:** Any general feedback/areas to improve?
+
+**Trevor:** I've been quite happy with the core functionality and API, as well as the support and community responsiveness on Slack and the forums. The minor issues I've had so far, from most important to least:
+
+* There is no official support for zero-downtime upgrades. Downtime causes us contractual headaches around SLAs and personal headaches around performing upgrades during the middle of the night on a weekend. I've done this successfully at my own risk, but it will make me increasingly nervous. This feature seems doable, even in the face of schema upgrades, with some constraints (both for FA and its users) on consecutive versions.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/_posts/2021-01-26-fitt-finder-fusionauth.md:74]: Thanks, filed an issue: https://github.com/FusionAuth/fusionauth-issues/issues/1240

File: site/_posts/2021-01-26-fitt-finder-fusionauth.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+* As a Java-based service, the memory demands are relatively high and insufficient memory results in a JVM crash. I know [more recent JVMs](https://medium.com/adorsys/usecontainersupport-to-the-rescue-e77d6cfea712) have better support for running in a container (i.e. no more `-Xmx`), so it would be nice to see FusionAuth support this (no more `FUSIONAUTH_MEMORY`) and degrade more gracefully (fail individual operations instead of the whole JVM).
+* Better clarity about what features need or benefit from Elasticsearch, so I can know whether I really need the expense and burden of running it. (Recommended production configurations start at around $400/mo on AWS.)
+* The FusionAuth UI is a bit dated looking and not always intuitive or ergonomic. For example, most of the useful operations on a table view involve tiny buttons on the right edge.
+* Make the FA core open source so users can better debug issues, more people can audit security, the community can work on issues that aren't a priority for the company, etc.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/_posts/2021-01-26-fitt-finder-fusionauth.md:78]: thanks, I linked to this: https://fusionauth.io/community/forum/topic/22/why-isn-t-fusionauth-open-source which also does a good job of presenting the tradeoffs.

File: site/_posts/2021-01-26-fitt-finder-fusionauth.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+**Dan:** Any general feedback/areas to improve?
+
+**Trevor:** I've been quite happy with the core functionality and API, as well as the support and community responsiveness on Slack and the forums. The minor issues I've had so far, from most important to least:
+
+* There is no official support for zero-downtime upgrades. Downtime causes us contractual headaches around SLAs and personal headaches around performing upgrades during the middle of the night on a weekend. I've done this successfully at my own risk, but it will make me increasingly nervous. This feature seems doable, even in the face of schema upgrades, with some constraints (both for FA and its users) on consecutive versions.
+* As a Java-based service, the memory demands are relatively high and insufficient memory results in a JVM crash. I know [more recent JVMs](https://medium.com/adorsys/usecontainersupport-to-the-rescue-e77d6cfea712) have better support for running in a container (i.e. no more `-Xmx`), so it would be nice to see FusionAuth support this (no more `FUSIONAUTH_MEMORY`) and degrade more gracefully (fail individual operations instead of the whole JVM).
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/_posts/2021-01-26-fitt-finder-fusionauth.md:75]: I think he's comparing the memory demands of FusionAuth to the memory demands of a binary only program like one built in golang. The article he references includes switches that are enabled by default in java 14. I will add a note about additional configuration options that might help.

File: site/_posts/2021-01-26-fitt-finder-fusionauth.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jobannon) [reply site/_posts/2021-01-26-fitt-finder-fusionauth.md]: Thanks for filing - I suppose that I could have taken the initiative too ... 🙂   That would be a cool feature huh?  👍🏻

**Pull request merged**